### PR TITLE
FileUploadWidget: Invalid prop definition for multiple uploads (#333)

### PR DESCRIPTION
**Pull request by martinfrances107**
## Issue

https://github.com/jsdrupal/drupal-admin-ui/issues/333

## Testing instructions

This reduces one of the many  warnings that appear on the console log when visiting 

for http://localhost:3000/node/add/recipe
[ The warnings were also visible when looking at the storybook definitions of the FileUploadWidget ]

The parent issue has a description of the bug. But here is a short summary of the solution.

There were two bugs one hiding behind the other

The first bug was a misconception... 

PropTypes.oneOf() incorrectly instead of PropTypes.oneOfType() 

The second bug was just to get the PropTypes to make sense

the id which is required is different in the two distinct cases.

### CODE REVIEW

**Comment by dawehner** [comment]
Thank you for fixing these prop types, I wish we would have a nicer language :)

Sorry there is a conflict now :(

**Comment by martinfrances107** [comment]
Ah no problem.. there was a flurry of commits within about 10mins .... but hey it is good to see problems go away. I will fix this up in the morning.

**Comment by martinfrances107** [comment]
Ok so this is now a follow-up issue

.. so I have rebased on the new master and forced a push...

It removes the error... but now FileUploadWidget can be parsed a little more, revealing a new set of error messages. These warnings are related to the bad 'key" structure of the List .. ListItem structure 

I will solve those bugs as part of 

#321 - FileUploadWidget: Adjust the semantic structure of

File: src/components/02_atoms/Widgets/FileUploadWidget.js [hunk 1]
```diff
@@ -262,23 +262,36 @@ class FileUploadWidget extends React.Component {
   }
 }
 
-const fileItemSchema = PropTypes.shape({
+const fileItemMultiplePropType = PropTypes.shape({
+  id: PropTypes.toString.isRequired,
+  file: PropTypes.shape({
```
Comment (dawehner) [inline src/components/02_atoms/Widgets/FileUploadWidget.js]: Should we at least share the file part of this proptype?

Comment (martinfrances107) [reply src/components/02_atoms/Widgets/FileUploadWidget.js]: Thanks, that is a good idea.

**Pull request merged**